### ISSUE TITLE
quaternion_to_axis_angle: wrong identity gradient for negative and non-unit quaternions

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4237]
`quaternion_to_axis_angle` returns a finite but incorrect gradient at identity quaternions other than `(1, 0, 0, 0)`. In particular, `(-1, 0, 0, 0)` is the same unit identity rotation but its vector-component derivatives have the wrong sign. Scaled identity inputs also have the wrong derivative magnitude; the function explicitly permits non-unit quaternions.

## Reproduction

```python
import torch
from kornia.geometry.conversions import quaternion_to_axis_angle

for w in (1.0, -1.0, 2.0, -2.0):
    q = torch.tensor([w, 0.0, 0.0, 0.0], dtype=torch.float64, requires_grad=True)
    jac = torch.autograd.functional.jacobian(quaternion_to_axis_angle, q)
    delta = torch.tensor([0.0, 1e-6, 0.0, 0.0], dtype=q.dtype)
    numerical = (
        quaternion_to_axis_angle(q + delta) - quaternion_to_axis_angle(q - delta)
    ) / (2e-6)
    print(w, jac[0, 1].item(), numerical[0].item())

q = torch.tensor([-1.0, 0.0, 0.0, 0.0], dtype=torch.float64, requires_grad=True)
torch.autograd.gradcheck(quaternion_to_axis_angle, (q,))  # Jacobian mismatch
```

Observed, rounded:

| Real component `w` | Autograd `d output_x / d input_x` | Central difference |
| --- | --- | --- |
| 1 | 2 | 2 |
| -1 | 2 | -2 |
| 2 | 2 | 1 |
| -2 | 2 | -1 |

`gradcheck` passes for `w=1` and fails for each of the other three inputs.

## Expected behavior and cause

For fixed nonzero `w` and a vector part approaching zero, the implemented surrounding map has first-order expansion `axis_angle = (2 / w) * vector_part`. The derivative at the zero vector part should agree with this limit. Both signs of the unit identity must work.

The zero-vector branch instead uses the constant `k_neg = 2.0 * torch.ones_like(sin_theta)` at [conversions.py](https://github.com/kornia/kornia/blob/3f102790/kornia/geometry/conversions.py#L982-L984), independently of `w`. This gives the correct value (zero) but the correct derivative only when `w=1`.

The NaN-gradient repair for #3949 made this branch finite, but the current `test_convention_gradients_at_the_identity_are_finite_3949` only tests the positive unit identity. Its analytic-gradient assertion therefore does not catch this case. The current convention block promises the analytic limit at identity.

## Impact and regression coverage

This is a backward-correctness defect: a valid unit quaternion can produce the opposite gradient direction in pose optimization even though the forward rotation is correct. It should be handled as a broken-output repair, with regression checks for both unit identity signs, non-unit identities, and mixed batches. The all-zero quaternion is a separate degenerate-input policy question; it is not needed to reproduce this issue.

## Verification

Reproduced on Kornia `0.9.0rc1`, commit `3f102790`, PyTorch `2.14.0`, Python `3.11.14`, macOS CPU, float64. Central differences and `torch.autograd.gradcheck` independently expose the mismatch. CUDA and MPS were not checked. The existing convention/wart selection across conversions, bbox/boxes, and geometry transforms remains green: 343 passed, 17 xfailed, 2 skipped on CPU float32/float64.

Posted on behalf of [@ducha-aiki](https://github.com/ducha-aiki) by Codex (gpt-6-astra).
